### PR TITLE
Update docusaurus to 2.0.0-beta.15

### PR DESCRIPTION
**Pull request by noahp**
A lot of changes since `-beta.0`. I'm pulling this in to resolve the
dependabot auto-PRs en masse.

Updated our TargetVersionFilled darkmode sensitive function with the API
changes in docusaurus.

Add a `.node-version` file indicating we need ^14 now.

Also update the .forestry image to a node v14 one (I don't know what
that's used for but 🤷)

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -175,6 +171,10 @@ module.exports = {
                     editUrl:
                         "https://github.com/memfault/memfault-docs/edit/main/",
                 },
+                gtag: {
+                    trackingID: "G-R5JYJ06TDJ",
```
Comment (noahp) [inline docusaurus.config.js:175]: TODO confirm that this ends up in the generated site!!!

Comment (noahp) [reply docusaurus.config.js:175]: Confirmed (I just ran `yarn build && rg --no-ignore -t html G-R5JYJ06TDJ`)

**Comment by noahp** [comment]
@fnune should that `.node-version` file be enough to convince netlify to use nodejs 14? seems to not be picking it up and I don't have access to the site settings :confused: https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-and-javascript

**Review by vlindhol — APPROVED** [review]
LGTM!

**Review by chrisc11 — APPROVED** [review]
Cool, I like some of the small style changes in the new version

> Also update the .forestry image to a node v14 one (I don't know what
that's used for but 🤷)

no need to do it in this PR but we aren't using it anymore so can be dropped fwiw

**Comment by noahp** [comment]
Did a pass through most of the pages, didn't see anything suspicious. Merging and let's keep an eye out for a revert.

**Pull request merged**